arangod/VocBase/Methods/Collections.cpp

```diff
@@ -892,6 +893,30 @@ futures::Future<Result> Collections::warmup(TRI_vocbase_t& vocbase,
   return futures::makeFuture(res);
 }
 
+futures::Future<Result> Collections::upgrade(TRI_vocbase_t& vocbase,
+                                             LogicalCollection const& coll) {
+  ExecContext const& exec = ExecContext::current();  // disallow expensive ops
+  if (!exec.canUseCollection(coll.name(), auth::Level::RO)) {
```

Comment:
Is read-only access sufficient for performing the upgrade? Or should we require read-write permissions? Not sure if the `RO` is intentional here or not. If so, a code comment would help.

Reply:
Ah, no, you are correct. This is a mistake.